### PR TITLE
modify syncd init script for supporting yml

### PR DESCRIPTION
**Pull request by geans-pin**

### CODE REVIEW

**Comment by kcudnik** [comment]
changes like this should be cherrypicked and have the same name as original PR not "new commit same as"

**Comment by geans-pin** [comment]
Any idea of the Azure Pipelines 13 ran longer than the maximum time of 60 mins ?

##[error]The job running on agent Azure Pipelines 13 ran longer than the maximum time of 60 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134
Agent: Azure Pipelines 13
Started: Today at 下午2:30
Duration: 1h 0m 7s

Job preparation parameters
9 queue time variables used
[1 artifact](https://dev.azure.com/mssonic/build/_build/results?buildId=707294&view=artifacts&pathAsName=false) produced

**Comment by kcudnik** [comment]
```
2024-11-28T06:54:18.5245096Z [       OK ] ClientServerSai.FDB_ENTRY (8 ms)
2024-11-28T06:54:18.5245628Z [ RUN      ] ClientServerSai.FLOW_ENTRY
2024-11-28T07:30:08.7223662Z ##[error]The Operation will be canceled. The next steps may not contain expected logs.
2024-11-28T07:30:08.7401297Z ##[error]The operation was canceled.
2024-11-28T07:30:08.7405562Z ##[section]Finishing: Run sonic sairedis unit tests
```

usually build process is les than 30 min, but here for somehow its stuck on FLOW_ENTRY,  you would nedd to download syslog from that build and analyse syslog what went wrong

**Comment by geans-pin** [comment]
With /azpw run, all checker pass

**Comment by Praveen-Brcm** [comment]
@ZhaohuiS Can you please help approve and merge the PR. All checks are passing now.

File: syncd/scripts/syncd_init_common.sh [hunk 1]
```diff
@@ -122,74 +122,188 @@ config_syncd_cisco_8000()
...
-          CMD_ARGS+=" -p /etc/sai.d/sai.profile"
-      else
-          CMD_ARGS+=" -p $HWSKU_DIR/sai.profile"
-      fi
+config_syncd_bcm()
+{
+    PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+    PLT_CONFIG_BCM=""
+    PLT_CONFIG_YML=""
+    PLT_SAI_PROFILE=$(find $HWSKU_DIR -name 'sai.profile')
```
Comment (ZhaohuiS) [inline syncd/scripts/syncd_init_common.sh:239]: @geans-pin sai.profile should located in `/etc/sai.d` folder
Take 6100 for example, sai.profile is not in `/usr/share/sonic/hwsku`


```
root@str3-s6100-acs-7:/etc/sai.d# cat sai.profile 
SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/th-s6100-64x40G-t1.config.bcm
SAI_NUM_ECMP_MEMBERS=64
root@str3-s6100-acs-7:/etc/sai.d# cd /usr/share/sonic/hwsku
root@str3-s6100-acs-7:/usr/share/sonic/hwsku# ls -al
total 41
drwxr-xr-x 5 root root   384 Mar 14  2024 .
drwxr-xr-x 1 root root  4096 Nov 27 02:49 ..
drwxr-xr-x 2 root root   123 Mar 14  2024 BALANCED
drwxr-xr-x 2 root root   123 Mar 14  2024 RDMA-CENTRIC
drwxr-xr-x 2 root root   123 Mar 14  2024 TCP-CENTRIC
-rw-r--r-- 1 root root   221 Feb 20  2017 buffer_ports_t0.j2
-rw-r--r-- 1 root root   221 Feb 20  2017 buffer_ports_t1.j2
-rw-r--r-- 1 root root    66 Feb 20  2017 buffers.json.j2
-rw-r--r-- 1 root root  1238 Feb 20  2017 buffers_defaults_t0.j2
-rw-r--r-- 1 root root  1238 Feb 20  2017 buffers_defaults_t1.j2
-rw-r--r-- 1 root root   812 Feb 20  2017 pg_profile_lookup.ini
-rw-r--r-- 1 root root  3953 Feb 20  2017 port_config.ini
-rw-r--r-- 1 root root   775 Feb 20  2017 qos.json.j2
-rw-r--r-- 1 root root   728 Feb 20  2017 sai.profile.j2
-rw-r--r-- 1 root root 10607 Feb 20  2017 th-s6100-64x40G-t0.config.bcm
-rw-r--r-- 1 root root 10650 Feb 20  2017 th-s6100-64x40G-t1.config.bcm
```

Comment (geans-pin) [reply syncd/scripts/syncd_init_common.sh]: I've changed the search path to /etc/sai.d  which should be common for all broadcom sonic platforms

**Comment by adyeung** [comment]
@ZhaohuiS @StormLiangMS please review and signoff if there is no further comments

**Review by anilkpan — APPROVED** [review]

File: syncd/scripts/brcm_common_config_ut.sh [hunk 2]
```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+###BRCM Common config UT####
+HWSKU_DIR=/usr/share/sonic/hwsku
+PLATFORM_COMMON_DIR=/usr/share/sonic/device/x86_64-broadcom_common
+PLT_SAI_PROFILE=$(find $HWSKU_DIR -name 'sai.profile')
```
Comment (ZhaohuiS) [inline syncd/scripts/brcm_common_config_ut.sh:5]: @geans-pin same sai.profile location issue for ut test.

**Comment by ZhaohuiS** [comment]
> @ZhaohuiS @StormLiangMS please review and signoff if there is no further comments

@adyeung  I found same sai.profile location issue for UT script. Could you please test it and verify it works then submit the commits? I spent days on testing, but obviously the test coverage on your side is not enough.

**Comment by geans-pin** [comment]
> > @ZhaohuiS @StormLiangMS please review and signoff if there is no further comments
> 
> @adyeung I found same sai.profile location issue for UT script. Could you please test it and verify if it works then submit the commits?

WIth or Without the sai.profile search path change for all the MICAS platforms we have,  the function and UT works well. 
Apply the same search path change to UT script in the new commit. Please help to verify on your specific platform with different location of sai.profile.

**Review by ZhaohuiS — APPROVED** [review]
Verified on 7260cx3/7060/7050cx3/s6100, it works.

**Review by kcudnik — APPROVED** [review]

**Pull request merged**